### PR TITLE
feat: JaxMultiStartOptimizer — vmap-fused multi-start inside XLA

### PR DESCRIPTION
**Pull request by ericchansen**
## Summary

Adds `JaxMultiStartOptimizer` — the vmap-fused version of the existing Python-loop `MultiStartOptimizer`. N replicas of a jaxopt solver run in parallel inside a single `jax.vmap`-compiled XLA kernel. The loss, gradients, L-BFGS line search, and best-of-N argmin selection all happen inside one JIT graph.

This is Phase 3 of the end-to-end analytical differentiation/optimization roadmap — pushing multi-start into the XLA graph so CH₃F-style multi-start sweeps can scale on GPU.

### New
- `q2mm/optimizers/jax_multistart.py` — `JaxMultiStartOptimizer`
  - Methods: `lbfgs` (default), `lbfgsb`, `gradient_descent`
  - Kwargs: `n_starts`, `maxiter`, `tol`, `perturbation_pct`, `seed`, `verbose`
  - First replica is unperturbed; remaining `n_starts-1` use `±perturbation_pct * |x0|`, clipped to bounds
  - `lbfgsb` inherits the existing GPU guard (raises `RuntimeError` on non-CPU; use `lbfgs` on GPU)
- `test/test_jax_multistart.py` — 12 tests: validation, H₂/water convergence, `n_starts=1` parity with plain `JaxOptOptimizer`, seed determinism, argmin correctness, forcefield update, LBFGSB GPU guard

### Wiring
- `q2mm/optimizers/__init__.py` — lazy export
- `q2mm/diagnostics/benchmark.py` — `jaxopt-multi:METHOD` dispatch mirroring `jaxopt:` and `multi:` branches

### Drive-by fix
- `test/test_jaxopt_optimizer.py` — the `test_lbfgs_h2_energy` assertion for `result.jac_mode == "jit"` was broken on master; production code emits `"analytical"`. Updated the test to match.

## Verification

- ruff lint + format clean
- 594 core tests pass (`-m "not (openmm or tinker or jax or jax_md or psi4)"`)
- 38 jax tests pass (`test/test_jaxopt_optimizer.py test/test_jax_loss.py test/test_jax_multistart.py`)
- Pre-commit ruff hook passed

## Roadmap context

Refs #176 (JAX-native multi-start was one of the phases implicitly bundled into that umbrella issue). Phase 2 (vmap over molecules) remains open — multi-start over starts is the orthogonal axis and doesn't block it.

### CODE REVIEW

File: test/test_jax_multistart.py [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+
+        mol = make_diatomic(distance=0.74, bond_tolerance=1.5)
+        ff = _h2_ff()
+        engine = JaxEngine()
+
+        ref = ReferenceData()
+        ref.add_energy(value=0.0, molecule_idx=0, weight=1.0)
+
+        return ObjectiveFunction(forcefield=ff, engine=engine, molecules=[mol], reference=ref)
+
```
[the remark this answers is not in the document]
Comment (ericchansen) [reply test/test_jax_multistart.py]: Fixed in 9eede5a. Annotated _make_h2_obj as returning ObjectiveFunction.

File: q2mm/optimizers/jax_multistart.py [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+            final_score=best_score,
+            n_iterations=self.maxiter,
+            n_evaluations=self.maxiter * self.n_starts,
+            initial_params=x0,
+            final_params=best_params,
+            history=[initial_score, best_score],
+            method=method_str,
+            jac_mode="analytical",
+            eps=None,
+        )
```
[the remark this answers is not in the document]
Comment (ericchansen) [reply q2mm/optimizers/jax_multistart.py:241]: Fixed in 9eede5a. success/n_iterations/message now derive from the best replica's jaxopt state (state_batch.error and state_batch.iter_num), matching JaxOptOptimizer.

File: q2mm/optimizers/jax_multistart.py [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+        if self.verbose:
+            logger.info(
+                "%s best: %.6f (replica %d/%d; scores min=%.6f, median=%.6f, max=%.6f)",
+                method_str,
+                best_score,
+                best_idx,
+                self.n_starts,
+                float(final_scores_np.min()),
+                float(np.median(final_scores_np)),
+                float(final_scores_np.max()),
```
[the remark this answers is not in the document]
Comment (ericchansen) [reply q2mm/optimizers/jax_multistart.py]: Fixed in 9eede5a. final_scores/final_params_batch stay on-device; only the best-replica slice (params + error + iter_num) and scalar min/median/max summaries are device_get'd.

File: q2mm/optimizers/jax_multistart.py [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+    are uniformly perturbed by ``±perturbation_pct`` of each parameter
+    value.
+
+    Args:
+        method: jaxopt method name.  One of ``'lbfgs'`` (default),
+            ``'lbfgsb'`` (CPU-only), or ``'gradient_descent'``.
+        n_starts: Number of parallel replicas.  Must be ``>= 1``.
+        maxiter: Maximum optimizer iterations per replica.
+        tol: Convergence tolerance (passed to the solver; not used for
+            early-stopping the vmapped run).
```
[the remark this answers is not in the document]
Comment (ericchansen) [reply q2mm/optimizers/jax_multistart.py]: Fixed in 9eede5a. Updated the tol docstring to reflect that each replica may stop early when tol is met.

File: q2mm/optimizers/jax_multistart.py [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+
+    def optimize(self, objective: ObjectiveFunction) -> OptimizationResult:
+        """Run ``n_starts`` replicas in parallel, return the best.
+
+        Args:
+            objective: Configured objective with a JaxEngine backend.
+
+        Returns:
+            OptimizationResult for the replica with the lowest final
+            loss.  ``extra`` carries per-replica final scores.
```
[the remark this answers is not in the document]
Comment (ericchansen) [reply q2mm/optimizers/jax_multistart.py]: Fixed in 9eede5a. Dropped the stale 'extra carries per-replica scores' line; docstring now accurately describes the return.

File: test/test_jax_multistart.py [hunk 6]
```diff
@@ -0,0 +1,286 @@
...
+            n_starts=5,
+            maxiter=50,
+            perturbation_pct=0.2,
+            seed=42,
+            verbose=False,
+        )
+        r_b = opt_b.optimize(obj_b)
+
+        np.testing.assert_allclose(r_a.final_params, r_b.final_params, rtol=1e-8)
+        assert r_a.final_score == r_b.final_score
```
[the remark this answers is not in the document]
Comment (ericchansen) [reply test/test_jax_multistart.py]: Fixed in 9eede5a. Switched to pytest.approx(abs=1e-8) for the determinism comparison.
